core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/MemorySailStore.java

```diff
@@ -569,13 +565,12 @@ public synchronized void deprecate(Resource subj, IRI pred, Value obj, Resource
 			}
 		}
 
-		private void acquireExclusiveTransactionLock()
+		private synchronized void acquireExclusiveTransactionLock()
```

Comment:
I synchronized this to bring it in line with the same method in NativeSailStore. You're right though that we should probably look at the calling public methods. I'll have a closer look and adapt where necessary.